### ISSUE TITLE
TensorFlow PointNet does not train on modelnet40_ply_hdf5_2048

### ISSUE THREAD

**Issue by siteh123** [fxia22/pointnet.pytorch#112]
Hello,

We are experiencing a critical issue when trying to train the original TensorFlow PointNet implementation on ModelNet40 using the modelnet40_ply_hdf5_2048 dataset.

**Environment:**
TensorFlow version: 2.x
GPU: NVIDIA GeForce RTX 4070 SUPER (CUDA detected)
Dataset path: data/modelnet40_ply_hdf5_2048

TensorFlow correctly detects and uses GPU.
Dataset shapes and label ranges are valid.

**Dataset:**
Files present: ply_data_train0.h5 … ply_data_train4.h5, ply_data_test0.h5, ply_data_test1.h5
Data shape confirmed from logs: (2048, 1024, 3) for most files; labels are integers 0–39.
Unique labels verified for each file; no missing classes.

**Behavior:**
Loss stays almost constant around 4.7 → 4.5 even after 35+ epochs.
Training accuracy remains extremely low, ~0.08–0.09 (random chance for 40 classes is 0.025).
Evaluation accuracy remains around 0.04, with average class accuracy 0.025.

Logs show minimal drift in loss due to batch shuffling, but no meaningful learning occurs.

> **** EPOCH 000 ****
> ----0-----
> Train file: data/modelnet40_ply_hdf5_2048/ply_data_train4.h5
> mean loss: 4.704373
> accuracy: 0.074755
> ...
> **** EPOCH 047 ****
> ----0-----
> Train file: data/modelnet40_ply_hdf5_2048/ply_data_train3.h5
> mean loss: 4.455300
> accuracy: 0.095215

Are there known issues with TensorFlow PointNet on this HDF5 PLY dataset?

Guidance on dataset preprocessing, training parameters, or potential bugs would be appreciated.
